mlflow/store/sqlalchemy_store.py

```diff
@@ -399,12 +400,20 @@ def set_tag(self, run_uuid, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_uuid, key=tag.key, value=tag.value))
 
-    def search_runs(self, experiment_ids, search_filter, run_view_type):
+    def search_runs(self, experiment_ids, search_filter, run_view_type, max_results):
+        # TODO: push search query into backend database layer
+        if max_results > SEARCH_MAX_RESULTS_THRESHOLD:
+            raise MlflowException("Search API called with a large max_results ({}). Max threshold "
```

Comment:
Wording nit/suggestion: `"Invalid value for request parameter max_results. max_results must be at most {}, but got value {}".format(SEARCH_MAX_RESULTS_THRESHOLD, max_results)`